yogstation/code/modules/power/NuclearEngine.dm

```diff
@@ -0,0 +1,458 @@
+	icon_state = "wastehatch-closed"
+
+/obj/effect/reactoroverlay/monitor
+	icon_state = "monitor-off"
+
+/obj/machinery/power/NuclearReactor/CtrlClick(mob/user)
+	if(state & FUELHATCH_OPEN)
+		state &= ~FUELHATCH_OPEN //Bitflags! We add the OPPOSITE of fuelhatch open to "state", in other words, inverting it, if the hatch is open, this closes it
+		playsound(loc, 'sound/effects/bin_close.ogg',50,1)
+		to_chat(user,"You shut the fuel hatch")
+	else
+		state |= FUELHATCH_OPEN
+		playsound(loc, 'sound/effects/bin_open.ogg',50,1)
+		to_chat(user,"You open the fuel hatch")
+	UpdateIcon()
+
+/obj/machinery/power/NuclearReactor/AltClick(mob/user)
+	if(state & WASTEHATCH_OPEN)
+		state &= ~WASTEHATCH_OPEN //Bitflags! We add the OPPOSITE of wastehatch open to "state", in other words, inverting it, if the hatch is open, this closes it
+		playsound(loc, 'sound/effects/bin_close.ogg',50,1)
+		to_chat(user,"You shut the access hatch")
+	else
+		state |= WASTEHATCH_OPEN
+		playsound(loc, 'sound/effects/bin_open.ogg',50,1)
+		to_chat(user,"You open the access hatch")
+	UpdateIcon()
+
+/obj/machinery/power/NuclearReactor/proc/UpdateIcon()
+	cut_overlays()
+	waste.icon_state = "wastehatch-closed"
+	fuel.icon_state = "hatch-closed"
+	if(state & FUELHATCH_OPEN)
+		fuel.icon_state = "hatch-open"
+	if(state & WASTEHATCH_OPEN)
+		waste.icon_state = "wastehatch-open"
+	if(Heat >= LOW_HEAT)
+		monitor.icon_state = "monitor-low"
+	if(Heat > MEDIUM_HEAT)
+		monitor.icon_state = "monitor-med"
+	if(Heat > HIGH_HEAT)
+		monitor.icon_state = "monitor-high"
+	if(Heat > CRITICAL_HEAT) //At this stage. Worry. A lot.
+		monitor.icon_state = "monitor-warning"
+	add_overlay(waste)
+	add_overlay(fuel)
+	add_overlay(monitor)
+
+/obj/machinery/power/NuclearReactor/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/twohanded/required/FuelRod))
+		if(state & FUELHATCH_OPEN)
+			to_chat(user, "You slot [I] into [src]!")
+			I.forceMove(src)
+			react.FRS += I
+		else
+			to_chat(user, "Open the fuel hatch first!")
+	if(istype(I, /obj/item/twohanded/required/ControlRod))
+		to_chat(user, "You slot [I] into [src]!")
+		I.forceMove(src)
+		react.CRS += I
+
+/obj/machinery/power/NuclearReactor/attack_hand(mob/user)
+	if(!ReactorInoperable)
+		if(state & WASTEHATCH_OPEN)
+			var/obj/C = input(user, "Remove what?", "Reactor Control Hatch") as null|obj in contents
+			C.forceMove(get_turf(user))
+			to_chat(user, "You carefully slide [C] out of [src]")
+			if(istype(C,/obj/item/twohanded/required/FuelRod))
+				react.FRS -= C
+			if(istype(C,/obj/item/twohanded/required/ControlRod))
+				react.CRS -= C
+			if(prob(20))
+				radiation_pulse(src, 5, 10)
+		else
+			start()
+	else
+		radiation_pulse(src, 100, 20)
+
+/obj/machinery/power/NuclearReactor/Initialize()
+	. = ..()
+	react.reactor = src
+	UpdateIcon()
+	state |= FUELHATCH_OPEN
+	state |= WASTEHATCH_OPEN
+	connect_to_network()
+//	outlet = new(src.loc)
+	var/obj/dummy = new(src.loc) //Handle placing pipes properly
```

Comment:
You can change all this to `outlet = locate(/obj/machinery/atmospherics/components/binary/pump/nuclear) in get_step(src, NORTH)`. You probably could even omit the path, locate should automatically find the correct thing.